mailbag/controller.py

```diff
@@ -112,4 +113,15 @@ def generate_mailbag(self):
                         writer = csv.writer(f)
                         writer.writerows(portion)
 
+        if self.args.compress and not self.args.dry_run:
+            log.info("Compressing Mailbag")
+            if self.args.compress=="tar":
+                shutil.make_archive(mailbag_dir, 'tar', mailbag_dir)
+            if self.args.compress== "zip":
+                shutil.make_archive(mailbag_dir, 'zip', mailbag_dir)
+            if self.args.compress == "tar.gz":
+                shutil.make_archive(mailbag_dir, 'gztar', mailbag_dir)
+            if mailbag_dir:
```

Comment:
We should check if the archive file exists, as this should only check if the path exists in memory. I would use `os.path.isfile` but you're going to have to get the correct extension.